src/Bicep.Core/TypeSystem/TypeAssignmentVisitor.cs

```diff
@@ -803,6 +801,11 @@ private TypeSymbol GetDeclaredTypeAndValidateDecorators(DecorableSyntax targetSy
         public override void VisitProviderDeclarationSyntax(ProviderDeclarationSyntax syntax)
             => AssignTypeWithDiagnostics(syntax, diagnostics =>
             {
+                if (syntax.IsSkipped)
```

Comment:
What is the user experience in this case? Would this result in the user seeing multiple diagnostics for one mistake?